### PR TITLE
Fixed the music so it plays combat music correctly.

### PR DESCRIPTION
**Pull request by onpon4**
There were a few parts to this. In a nutshell:

1. Fixed the way enemies are counted so that disabled pilots are properly
   accounted for (previously a disabled and then destroyed enemy would
   decrease the enemy counter twice, for instance).

2. Removed the unnecessary requirement for the enemy becoming hostile
   to be the first enemy before setting combat music. (Lua already takes
   care of this, so the check in C was redundant.)

3. Fixed choose_combat Lua function so that if it finds that it's already
   playing, it still marks itself as the last music played (important because
   otherwise "ambient" is always set to "last" when jumping in, and music for
   missions ends up getting set before "idle" plays but after "ambient";
   if previous system had been playing combat music it would catch this and
   then not record itself as the "last" type, leading to ambient being "last"
   even though combat really was).

TL;DR: combat music now plays correctly, 100% of the time. No strange cases
of ambient music when you're fighting a gigantic horde of pirates.

### CODE REVIEW

**Comment by bobbens** [comment]
Looks good, thanks!
